protocol/blockchain/script.md

```diff
@@ -125,6 +125,8 @@ Op codes marked with **(do not use)** are disallowed and will make a transaction
 
 ### Arithmetic
 
+Numeric opcodes (OP_1ADD, etc) are restricted to operating on 4-byte integers. The semantics are subtle, though: operands must be in the range [-2^31 +1...2^31 -1], but results may overflow (and are valid as long as they are not used in a subsequent numeric operation). CScriptNum enforces those semantics by storing results as an int64 and allowing out-of-range values to be returned as a vector of bytes but throwing an exception if arithmetic is done or the result is interpreted as an integer.
```

Comment:
The last part of this looks language specific and probably should not be included.

`numeric operation` could also probably more precisely be stated with opcodes again. Together, you might use text like this:

> Numeric opcodes (OP_1ADD, etc) are restricted to operating on 4-byte integers. The semantics are subtle, though: operands must be in the range [-2^31 +1...2^31 -1], but results may overflow (and are valid as long as they are not subsequently used with numeric opcodes).

Reply:
I can see that the text was updated with your feedback, resolving thread